packages/connector-thegraph/src/index.ts

```diff
@@ -21,48 +37,52 @@ class ConnectorTheGraph implements ConnectorInterface {
     // this.#appClient = createClient({ url: appSubgraphUrl('app_id') })
   }
 
-  async permissions(orgAddress: string): Promise<Permission[]> {
-    const query = `
-      query {
-        organization(id: "${orgAddress}") {
-          acl {
-            permissions {
-              entity
-              app {
-                address
-              }
-              role {
-                name
-                manager
-              }
-            }
-          }
-        }
-      }
-    `
-
-    const res = await this.#daoClient.query(query).toPromise()
-
-    const { permissions } = res?.data?.organization.acl
-    if (!permissions) {
-      return []
-    }
-
-    return permissions.map(
-      ({
-        app,
-        entity,
-        role,
-      }: {
-        app: any
-        entity: string
-        role: { name: string }
-      }) => ({
-        app: app?.address || '',
-        entity,
-        role: role.name,
-      })
-    )
+  async permissionsForOrg(orgAddress: string): Promise<Permission[]> {
+    const org = (await this._performQuery(
+      queries.ORGANIZATION_PERMISSIONS,
+      { orgAddress }
+    )).organization as OrganizationDataGql
+
+    return parsePermissions(this, org.acl?.permissions)
+  }
+
+  async appsForOrg(orgAddress: string): Promise<App[]> {
+    const org = (await this._performQuery(
+      queries.ORGANIZATION_APPS,
+      { orgAddress }
+    )).organization as OrganizationDataGql
+
+    return parseApps(this, org.apps)
+  }
+
+  async appByAddress(appAddress: string): Promise<App> {
+    const app = (await this._performQuery(
+      queries.APP_BY_ADDRESS,
+      { appAddress }
+    )).app as AppDataGql
+
+    return parseApp(this, app)
+  }
+
+  async repoForApp(appAddress: string): Promise<Repo> {
+    const app = (await this._performQuery(
+      queries.REPO_BY_APP_ADDRESS,
+      { appAddress }
+    )).app as AppDataGql
+
+    return parseRepo(this, app.repo)
+  }
+
+  private async _performQuery(query: DocumentNode, vars: any =  {}): Promise<any> {
```

Comment:
nitpick: should we use standard private methods? It could also remove the need to prefix with an underscore :-)

```suggestion
  async #performQuery(query: DocumentNode, vars: any =  {}): Promise<any> {
```

Reply:
I tried, but it doesn't seem to be an option with the current compiler: https://github.com/microsoft/TypeScript/issues/37677